src/Microsoft.OData.Core/UriParser/Resolver/ODataUriResolver.cs

```diff
@@ -290,6 +311,19 @@ public virtual IEnumerable<IEdmOperationImport> ResolveOperationImports(IEdmMode
                 return results;
             }
 
+            if (model.IsImmutable())
+            {
+                NormalizedModelElementsCache cache = GetNormalizedModelElementsCache(model);
+                IEnumerable<IEdmOperationImport> cachedResults = cache.FindOperationImports(identifier);
+
+                if (cachedResults != null)
+                {
+                    return cachedResults;

```

Comment:
why don't you do the checks for multiple operations here like you doing with navigation sources?

Reply:
Because this cache-based code path aims to replicate the existing behaviour of the `ResolveOperationImports` method, which is to return all the matches that were found.

Unlike the `FindNavigationSource` method, this method returns a collection. Keep in mind that the `IEdmModel` can have multiple operation overloads with the same name but different parameters. So this is handled differently from navigation sources.

Reply:
how do you check for the params? or when two operations with the same name are returned but with different params how do you know which is which?

Reply:
I don't know exactly. That is outside the scope of the cache, and possibly resolved elsewhere. The cache was only concerned in replicating the already existing behaviour in a more efficient manner.

Here's a snippet of the existing implementation (this code path will still be used when the model is not immutable):
```c#
return container.Elements.OfType<IEdmOperationImport>()
                .Where(source => string.Equals(identifier, source.Name, StringComparison.OrdinalIgnoreCase));
```
It would be a bug if the cache returned something different from the non-cached version.